File: sorts/insertion.go
package sorts

//InsertionSort []int
func InsertionSort(arr []int) {
	for i := 1; i < len(arr); i++ {
		o, j := arr[i], i-1
		for j >= 0 && arr[j] > o {
			arr[j+1] = arr[j]
			j--
		}
		arr[j+1] = o
	}
}

File: sorts/Insertion.java
package sorts;

public class Insertion {

	public static void sort(int[] arr) {
		for (int i = 1; i < arr.length; i++) {
			int o = arr[i];
			int j = i - 1;
			while (j >= 0 && arr[j] > o) {
				arr[j + 1] = arr[j];
				j--;
			}
			arr[j + 1] = o;
		}
	}

}
